### DISCUSSION TITLE
A way to track zoom on another monitor?

### DISCUSSION THREAD

**Discussion by sargeonline** [tryptech/obs-zoom-and-follow#17] [Q&A]
Just wondering if there was a way to track movement outside the primary monitor? 
I have a 3 monitor set up and would like to track a program and movement on one of the other monitors. I thought adjusting the 'Select monitor to use base for zooming' to another display would work but it doesn't seem to change any behavior - it still uses the primary monitor.

![image](https://user-images.githubusercontent.com/12741015/117259879-d3182d00-ae91-11eb-85b3-eaccf4d8775d.png)

**Comment by tryptech** [comment]
It should be using the other monitor, and then you just have to set up a display capture to use with it. I'll have to take a look at it at some point. I usually only use it on my primary monitor so I might have overlooked something.

**Reply by sargeonline** [reply]
That would be great. Have set it up as you outlined but it doesn't seem to track horizontally on the 2nd monitor.

**Comment by ngaerlan** [comment]
following

**Comment by ngaerlan** [comment]
I think this might be relevant.  What I notice is that after changing the transform information for my second monitor (non-primary monitor) when I zoom in and then track the mouse it will correctly track top to bottom but not left to right.  What happens is that there is no tracking until my mouse is all way into my primary monitor on the opposite side.  I believe this has to do with how OBS treats multiple monitors.  If you look at them there is a global location for each monitor that uses the Top/Left as the anchor point.  "Monitor 1 [resolution] @ [global coordinate]"  when using the primary monitor that global coordinate is 0,0, but I have a monitor to the right of my main monitor and its global coordinate is "3840,0".  Because I'm using two 4k monitors this simply means that the top/left global anchor for my second monitor starts 3840 pixels to the right of the primary monitor, but that the horizontal position can be the same.  Maybe this will lead to a solution or not? 

![Screenshot 2021-05-12 174520](https://user-images.githubusercontent.com/31300899/118061558-ff2c2600-b349-11eb-8a51-02cc0f248279.png)

**Reply by sargeonline** [reply]
Is there a way to update the global coordinate in the code for obs zoom and follow? I tried updating z_x = 1920 but it didn't seem to change anything.

**Comment by thepragmatic15** [comment]
I'm going to follow this discussion, because I would like to be able to apply the zoom to two screens, without having to go back to the settings to change the Zoom Source.
I'm using the Advanced Scene Switcher tool with the region parameter, which allows OBS to automatically record the screen my mouse is on.
I'll do some testing on my end, and if I find a solution, I'll post it in this thread.
The 1st thing I will try is to copy this script and save it under a new name by adding (v2) to the original name.
Then I will add this 2nd script to OBS, indicating the 2nd screen as (Source Screen).
Hope this trick works.
